src/wp-includes/media.php

```diff
@@ -989,7 +996,7 @@ function wp_get_attachment_image_src( $attachment_id, $size = 'thumbnail', $icon
 	 *                                    an array of width and height values in pixels (in that order).
 	 * @param bool         $icon          Whether the image should be treated as an icon.
 	 */
-	return apply_filters( 'wp_get_attachment_image_src', $image, $attachment_id, $size, $icon );
+	return apply_filters( 'wp_get_attachment_image_src', $image, $attachment_id, $size, $args );
```

Comment:
This is a breaking change and therefore must not be made. Let's keep passing the value of `$icon` here (a boolean). If `$args['icon']` is not set, that would simply be `false`.